src/pages/algorithms/BitAlgorithms/index.md

```diff
@@ -0,0 +1,55 @@
+
+We all know that 1 byte comprises of 8 bits and any integer or character can be represented using bits in computers, which we call its binary form(contains only 1 or 0) or in its base 2 form.
+Example:
+1) 14 = {1110 }2
+= 1 * 23 + 1 * 22 + 1 * 21 + 0 * 20
+= 14.
+
+2) 20 = {10100 }2
+= 1 * 24 + 0 * 23 + 1 * 22 + 0 * 21 + 0 * 20
+= 20.
+
+For characters, we use ASCII representation, which are in the form of integers which again can be represented using bits as explained above.
+
+Bitwise Operators:
+
+There are different bitwise operations used in the bit manipulation. These bit operations operate on the individual bits of the bit patterns. Bit operations are fast and can be used in optimizing time complexity. Some common bit operators are:
+
+NOT ( ~ ): Bitwise NOT is an unary operator that flips the bits of the number i.e., if the ith bit is 0, it will change it to 1 and vice versa. Bitwise NOT is nothing but simply the one’s complement of a number. Lets take an example.
+N = 5 = (101)2
+~N = ~5 = ~(101)2 = (010)2 = 2
+
+AND ( & ): Bitwise AND is a binary operator that operates on two equal-length bit patterns. If both bits in the compared position of the bit patterns are 1, the bit in the resulting bit pattern is 1, otherwise 0.
+A = 5 = (101)2 , B = 3 = (011)2 A & B = (101)2 & (011)2= (001)2 = 1
+
+OR ( | ): Bitwise OR is also a binary operator that operates on two equal-length bit patterns, similar to bitwise AND. If both bits in the compared position of the bit patterns are 0, the bit in the resulting bit pattern is 0, otherwise 1.
+A = 5 = (101)2 , B = 3 = (011)2
+A | B = (101)2 | (011)2 = (111)2 = 7
+
+XOR ( ^ ): Bitwise XOR also takes two equal-length bit patterns. If both bits in the compared position of the bit patterns are 0 or 1, the bit in the resulting bit pattern is 0, otherwise 1.
+A = 5 = (101)2 , B = 3 = (011)2
+A ^ B = (101)2 ^ (011)2 = (110)2 = 6
+
+Left Shift ( << ): Left shift operator is a binary operator which shift the some number of bits, in the given bit pattern, to the left and append 0 at the end. Left shift is equivalent to multiplying the bit pattern with 2k
+
+( if we are shifting k bits ).
+1 << 1 = 2 = 21
+1 << 2 = 4 = 22 1 << 3 = 8 = 23
+1 << 4 = 16 = 24
+…
+1 << n = 2n
+
+Right Shift ( >> ): Right shift operator is a binary operator which shift the some number of bits, in the given bit pattern, to the right and append 1 at the end. Right shift is equivalent to dividing the bit pattern with 2k ( if we are shifting k bits ).
+4 >> 1 = 2
+6 >> 1 = 3
+5 >> 1 = 2
+16 >> 4 = 1 
+
+![Bit-wise Operators](https://www.google.co.in/url?sa=i&rct=j&q=&esrc=s&source=images&cd=&cad=rja&uact=8&ved=&url=https%3A%2F%2Fwww.go4expert.com%2Farticles%2Fbitwise-operators-t29233%2F&psig=AOvVaw2L-3j7Ch6Mh_gHR1biTn4F&ust=1508842706916075)
```

Comment:
Can you please fix this URL as it's not rendering the image.

When you view this document there is no image.